src/main/java/org/frankframework/http/HttpSenderCached.java

```diff
@@ -0,0 +1,631 @@
+
+		final String filenameToUse;
+		if(isFile || StringUtils.isNotBlank(filename)) {
+			String filenamebackup = StringUtils.isBlank(part) ? partSessionKey : part;
+			filenameToUse = StringUtils.isNotBlank(filename) ? filename : filenamebackup;
+		} else {
+			filenameToUse = null;
+		}
+
+		String partname = isFile || StringUtils.isBlank(part) ? partSessionKey : part;
+		return FormBodyPartBuilder.create(partname, new MessageContentBody(partObject, mimeType, filenameToUse)).build();
+	}
+
+	@Override
+	protected Message extractResult(URI targetURI, HttpResponseHandler responseHandler, PipeLineSession session) throws SenderException, IOException {
+		int statusCode = responseHandler.getStatusLine().getStatusCode();
+
+		if (!validateResponseCode(statusCode)) {
+			Message responseBody = responseHandler.getResponseMessage();
+			String body = "";
+			if(responseBody != null) {
+				responseBody.preserve();
+				try {
+					body = responseBody.asString();
+				} catch(IOException e) {
+					body = "(" + ClassUtils.nameOf(e) + "): " + e.getMessage();
+				}
+			}
+			log.warn(getLogPrefix() + "httpstatus [" + statusCode + "] reason [" + responseHandler.getStatusLine().getReasonPhrase() + "]");
+			return new Message(body);
+		}
+
+		Message responseMessage = responseHandler.getResponseMessage();
+		if (!Message.isEmpty(responseMessage)) {
+			responseMessage.closeOnCloseOf(session, this);
+		}
+
+		if (responseHandler.isMultipart()) {
+			return handleMultipartResponse(responseHandler, session);
+		} else {
+			return getResponseBody(targetURI, responseHandler, statusCode);
+		}
+	}
+
+	public Message getResponseBody(URI targetURI, HttpResponseHandler responseHandler, int statusCode) throws IOException {
+		Header[] headers = responseHandler.getAllHeaders();
+		if (getHttpMethod() == HttpMethod.HEAD) {
+			XmlBuilder headersXml = new XmlBuilder("headers");
+			for (Header header : headers) {
+				XmlBuilder headerXml = new XmlBuilder("header");
+				headerXml.addAttribute("name", header.getName());
+				headerXml.setCdataValue(header.getValue());
+				headersXml.addSubElement(headerXml);
+			}
+			return Message.asMessage(headersXml.toXML());
+		}
+
+		super.setCache();
+
+		etagCache = super.getEtagCache();
+		messageCache = super.getMessageCache();
+		Message responseMessage = responseHandler.getResponseMessage();
+
+		// Begin Etag handling
+		Header etagHeader = null;
+
+		for (Header header : headers) {
+			String key = header.getName().toLowerCase().strip();
+			if (new String(key).toLowerCase().equals("etag")) {
+				etagHeader = header;
+			}
+		}
+
+		// If the statusCode is 304 and Etag is present
+		if (statusCode == 304 && etagHeader != null) {
+			responseMessage = new Message(messageCache.get(etagHeader.getValue()).getObjectValue().toString());
+			log.warn("LOGHIT: ");
+			log.warn(responseMessage);
+		}
+		// Etag is present but no 304
+		else if (etagHeader != null) {
+			log.warn("LOGTAG:");
+			log.warn(targetURI.toString());
```

Comment:
Should be 1 log entry to debug log instead and with 'logPrefix'. Could also be a bit more descriptive. "Caching response with ETag: [{etagHeader.getValue()}] for url: [{targetURI.toString()}]."

example: log.debug(logPrefix + "copied response body input stream [" + is + "] to output stream [" + outputStream + "]");